app/src/main/java/org/mozilla/fenix/share/listadapters/AppShareAdapter.kt

```diff
@@ -23,19 +28,24 @@ class AppShareAdapter(
         return AppViewHolder(view, interactor)
     }
 
-    override fun getItemCount(): Int = applications.size
-
     override fun onBindViewHolder(holder: AppViewHolder, position: Int) {
-        holder.bind(applications[position])
+        holder.bind(getItem(position))
     }
 
     fun updateData(applications: List<AppShareOption>) {
```

Comment:
We should just use `submitList` instead, no need to keep the `updateData` alias

Reply:
:+1: